From ce9533088c841c21bc762404868d361771796192 Mon Sep 17 00:00:00 2001
From: Luciano Santos <luc14n0@opensuse.org>
Date: Tue, 18 Oct 2022 04:12:31 -0300
Subject: [PATCH] Add man subdir to top-level meson.build

Add the forgotten 'man' subdir to top-level meson.build script and add
a boolean option to meson_options.txt so people can choose whether or
not to install man pages.
---
 man/meson.build   | 6 +++++-
 meson.build       | 3 +++
 meson_options.txt | 3 +++
 3 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/man/meson.build b/man/meson.build
index 6f57501..04951a0 100644
--- a/man/meson.build
+++ b/man/meson.build
@@ -1 +1,5 @@
-install_man('eiciel.1')
+want_man = get_option('man')
+
+if want_man == true
+  install_man('eiciel.1')
+endif
diff --git a/meson.build b/meson.build
index 5a7c7f6..e52a497 100644
--- a/meson.build
+++ b/meson.build
@@ -81,6 +81,9 @@ subdir('img')
 # Documentation
 subdir('doc')
 
+# Manpage(s)
+subdir('man')
+
 # GNOME post-install duties
 gnome.post_install(gtk_update_icon_cache : true)
 
diff --git a/meson_options.txt b/meson_options.txt
index c32b6b9..caca81b 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,3 +1,6 @@
+option('man', type : 'boolean',
+    value : 'false',
+    description : 'Install man page(s)')
 option('user-attributes',
     type : 'feature',
     value : 'enabled',
-- 
2.37.3

